src/librustc_trans/back/link.rs

```diff
@@ -190,6 +189,7 @@ pub fn link_binary(sess: &Session,
     let mut out_filenames = Vec::new();
     for &crate_type in sess.crate_types.borrow().iter() {
         // Ignore executable crates if we have -Z no-trans, as they will error.
+        // TODO do we need to check for CrateTypeMetadata here?
```

Comment:
Presumably `CrateTypeMetadata` halts before the linking stage, right? So we should just skip it here?